api/src/main/java/org/openmrs/api/impl/PatientServiceImpl.java

```diff
@@ -116,14 +117,16 @@ public void onShutdown() {
 	/**
 	 * @see org.openmrs.api.PatientService#savePatient(org.openmrs.Patient)
 	 */
+	@Transactional(isolation = Isolation.SERIALIZABLE)
 	@Override
 	public Patient savePatient(Patient patient) throws APIException {
+		synchronized(this) {
 		requireAppropriatePatientModificationPrivilege(patient);
```

Comment:
Did you intentionally decide not to TAB the code within the synchronised phrase?